internal/nginx/config/generator.go

```diff
@@ -9,8 +9,10 @@ import (
 // Generator generates NGINX configuration.
 // This interface is used for testing purposes only.
 type Generator interface {
-	// Generate generates NGINX configuration from internal representation.
-	Generate(configuration dataplane.Configuration) []byte
+	// GenerateHTTPConf generates NGINX HTTP configuration from internal representation.
+	GenerateHTTPConf(configuration dataplane.Configuration) []byte
+	// GenerateMainConf generates the main nginx.conf file.
+	GenerateMainConf(configGeneration int) []byte
```

Comment:
@pleshakov I added this method because the agent requires the `nginx.conf` to be a part of every config payload, otherwise, it errors and does not apply the config.